### DISCUSSION TITLE
change sort in the file picker

### DISCUSSION THREAD

**Discussion by BrianHicks** [helix-editor/helix#3841] [Ideas]
Hello! I'm exploring Helix (coming from Kakoune) and I'd like a way to change the ordering of the files in the file explorer. In my hand-rolled stuff for Kakoune (and before that in the Vim fzf plugin) I had a custom sorter that sorted by edit distance from the current file. I asked on Matrix and it sounds like there is not already a way to do this.

Here's why I'd like this: when I'm doing stuff in my work monorepo, I frequently want to find related files first, and they tend to be named similarly. So a fuzzy search on `README.md` (which we have a lot of) surfacing something nearer to the current file in the filesystem hierarchy is useful. Same with finding tests: with a edit-distance sort, I can usually type `spec` or `test` (depending on the language) and the relevant file shows up quickly.

I already implemented a little pipe-y program for sorting the files at https://git.bytes.zone/brian/similar-sort. Would this be a nice addition to Helix, or would it be better in a plugin (once those exist?)

**Comment by cd-a** [comment]
The sorting in the file picker is an issue for me as well, let me explain:

This is how it's set up in neovim:
<img width="736" alt="CleanShot 2022-10-11 at 08 03 31@2x" src="https://user-images.githubusercontent.com/75008413/195009952-82e073e9-42ae-4774-8a0a-5c8f25adcc33.png">


This is the current default on helix:
<img width="856" alt="CleanShot 2022-10-11 at 08 03 51@2x" src="https://user-images.githubusercontent.com/75008413/195009956-1ed158f8-a350-41cf-9b6c-3394146a3f3e.png">


I search by filename, and prefer the results on the files instead of folders. On neovim, I use the [https://github.com/natecraddock/telescope-zf-native.nvim](https://github.com/natecraddock/telescope-zf-native.nvim) plugin, which basically just implements [zf](https://github.com/natecraddock/zf). This prioritises matches on filenames, which I find way more useful.

Now in helix I have to type **app.t** to get where I want.

Is this something that is customisable or open to change in the future?

**Comment by johan-lejdung** [comment]
![Screenshot 2025-02-10 at 15 46 24](https://github.com/user-attachments/assets/3f76c563-0af7-4d8b-90a8-e53fdbdcb316)

Here is another good example. In VSCode I has it ordered in numerical descending order. It would be great if we could configure this somehow
